### PR TITLE
Add `eslint-plugin-jsx-a11y` and `eslint-plugin-react-hooks` to `react` config

### LINKED ISSUE #824: [react] Extend eslint-plugin-react-hooks and eslint-plugin-jsx-a11y
The Square React ESLint configs should be aware of the [Rules of Hooks](https://reactjs.org/docs/hooks-rules.html) and common accessibility rules when building React apps. Common issues with React hooks are missing dependencies in hooks, and conditionally calling hooks. Common accessibility issues in React are missing `alt` attributes on images, and missing labels on input fields.

We can use two ESLint plugins to help with these:
- [eslint-plugin-react-hooks](https://github.com/facebook/react/tree/main/packages/eslint-plugin-react-hooks) - Adds rules to ESLint to ensure the rules of hooks are enforced
- [eslint-plugin-jsx-a11y](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y) - Adds rules to ESLint to call out common accessibility mistakes

This change can be made to the base `react` config here: https://github.com/square/eslint-plugin-square/blob/master/lib/config/react.js#L6

I'm happy to make a PR for this change if others feel these rules are useful.

### PR DESCRIPTION
**Pull request by atdrago**
This PR adds `eslint-plugin-jsx-a11y` and `eslint-plugin-react-hooks` to the list of configs that the `react` config extends from

Resolves #824

### CODE REVIEW

File: lib/config/react.js [hunk 1]
```diff
@@ -3,7 +3,12 @@
 'use strict';
 
 module.exports = {
-  extends: [require.resolve('./base'), 'plugin:react/recommended'],
+  extends: [
+    require.resolve('./base'),
+    'plugin:react/recommended',
+    'plugin:jsx-a11y/recommended',
+    'plugin:react-hooks/recommended',
```
Comment (bmish) [inline lib/config/react.js:10]: Let's alphabetize this list.

**Comment by bmish** [comment]
In the README next to the `react` config, can we add a list of the plugins it extends, just like with the other configs.

File: README.md [hunk 2]
```diff
@@ -73,7 +73,7 @@ If you prefer not to adopt a specific rule, you can disable it:
 | --- | --- | --- |
 | | [base](lib/config/base.js) | Rules and configuration for any JavaScript-based project. Includes recommended and optional rules from [eslint], [prettier], [eslint-plugin-eslint-comments], [eslint-plugin-import], [eslint-plugin-unicorn], and more. |
 | 🔥 | [ember](lib/config/ember.js) | [Ember.js](https://www.emberjs.com/)-specific additions on top of `base`. Includes recommended and optional rules from [eslint-plugin-ember], kebab-case filename enforcement with [eslint-plugin-filenames], and more. |
-| | [react](lib/config/react.js) | [React](https://reactjs.org/)-specific additions on top of `base`. |
+| | [react](lib/config/react.js) | [React](https://reactjs.org/)-specific additions on top of `base`. Includes recommended rules from [eslint-plugin-jsx-a11y] and [eslint-plugin-react-hooks]. |
```
[the remark this answers is not in the document]
Comment (bmish) [reply README.md:76]: and alphabetical

[the remark this answers is not in the document]
Comment (atdrago) [reply README.md]: Good catch! Should be good to go now.

**Review by bmish — APPROVED** [review]
Thanks!

**Pull request merged**